ui/component/or-rules/src/json-viewer/or-rule-action-attribute.ts

```diff
@@ -107,52 +122,81 @@ export class OrRuleActionAttribute extends translate(i18next)(LitElement) {
-        let attributes: [string, string][] = [];
-
-        if (asset && asset.attributes) {
-            attributes = Object.values(asset.attributes)
-                .map((attr) => {
-                    const label = Util.getAttributeLabel(attr, descriptors[0], assetType, false);
-                    return [attr.name!, label];
-                });
-        } else if (assetDescriptor) {
-            const assetTypeInfo = AssetModelUtil.getAssetTypeInfo(assetDescriptor);
-
-            attributes =
-                !assetTypeInfo || !assetTypeInfo.attributeDescriptors
-                    ? []
-                    : assetTypeInfo.attributeDescriptors.map((ad) => {
-                        const label = Util.getAttributeLabel(ad, descriptors[0], assetType, false);
-                        return [ad.name!, label];
-                    });
-        }
-
-        attributes.sort(Util.sortByString((attr) => attr[1]));
-
-        let attributeInput: TemplateResult | undefined;
-
-        if (this.action.attributeName) {
-            const label = descriptors[1] && (descriptors[1].name === WellknownValueTypes.BOOLEAN) ? "" : i18next.t("value");
-            let inputType;
-            if(descriptors[0]?.format?.asSlider) inputType = InputType.NUMBER;
-            attributeInput = html`<or-attribute-input ?compact=${descriptors[1] && (descriptors[1].name === WellknownValueTypes.GEOJSONPOINT)} .inputType="${ifDefined(inputType)}" @or-attribute-input-changed="${(ev: OrAttributeInputChangedEvent) => this.setActionAttributeValue(ev.detail.value)}" .customProvider="${this.config?.inputProvider}" .label="${label}" .assetType="${assetType}" .attributeDescriptor="${descriptors[0]}" .attributeValueDescriptor="${descriptors[1]}" .value="${this.action.value}" .readonly="${this.readonly || false}"></or-attribute-input>`;
-        }
+        let optionsProvider: (search?: string) => Promise<[any, string][]>;
 
         return html`
-            <or-mwc-input id="matchSelect" class="min-width" .label="${i18next.t("asset")}" .type="${InputType.SELECT}" @or-mwc-input-changed="${(e: OrInputChangedEvent) => this._assetId = (e.detail.value)}" .readonly="${this.readonly || false}" .options="${idOptions}" .value="${idValue}"></or-mwc-input>
-            ${attributes.length > 0 ? html`
-                <or-mwc-input id="attributeSelect" class="min-width" .label="${i18next.t("attribute")}" .type="${InputType.SELECT}" @or-mwc-input-changed="${(e: OrInputChangedEvent) => this.setActionAttributeName(e.detail.value)}" .readonly="${this.readonly || false}" .options="${attributes}" .value="${this.action.attributeName}"></or-mwc-input>
-                ${attributeInput}
-            ` : html`
-                <or-translate value="No attributes with write permission"></or-translate>
-            `}
+            
+            <!-- Show SELECT input with 'loading' until the assets are retrieved -->
+            ${when((!this._assets), () => html`
```

Comment:
Here all the initial async options gathering is done at this higher level which comes back to my point about handling it at a higher level rather than in the `or-mwc-input` component